### PR TITLE
Change generic-data-science-notebook refs to red-hat-data-services

### PR DESCRIPTION
**Pull request by tmckayus**
Use red-hat-data-services repos as the source for generic data
science notebook builds and source code.

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-973
- [x] The Jira story is acked
- [x] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

### CODE REVIEW

**Comment by tmckayus** [comment]
The "rhods-version" name for the tag is a placeholder, will be updated via sed at imagebuild time for the operator

File: jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml [hunk 1]
```diff
@@ -15,11 +15,11 @@ spec:
...
-      opendatahub.io/notebook-python-dependencies: '[{"name":"Boto3","version":"1.17.11"},{"name":"Kafka-Python","version":"2.0.2"},{"name":"Matplotlib","version":"3.4.2"},{"name":"Numpy","version":"1.21.0"},{"name":"Pandas","version":"1.2.5"},{"name":"Scikit-learn","version":"0.24.2"},{"name":"Scipy","version":"1.7.0"}]'
-      openshift.io/imported-from: quay.io/thoth-station/s2i-generic-data-science-notebook
+      opendatahub.io/notebook-python-dependencies: '[{"name":"Boto3","version":"1.17.11"},{"name":"Kafka-Python","version":"2.0.2"},{"name":"Matplotlib","version":"3.4.1"},{"name":"Numpy","version":"1.19.2"},{"name":"Pandas","version":"1.2.4"},{"name":"Scikit-learn","version":"0.24.1"},{"name":"Scipy","version":"1.6.2"}]'
+      openshift.io/imported-from: quay.io/modh/odh-generic-data-science-notebook
     from:
       kind: DockerImage
-      name: quay.io/thoth-station/s2i-generic-data-science-notebook:v0.0.5
-    name: "v0.0.5"
+      name: quay.io/modh/odh-generic-data-science-notebook:nb-1
+    name: "v1.2.0-2"
```
Comment (tmckayus) [inline jupyterhub/notebook-images/overlays/additional/generic-data-science-notebook-imagestream.yaml:23]: oops, that's supoosed to be "rhods-version" (pushed the result of an experiment)

**Review by harshad16 — COMMENTED** [review]
lgtm :+1:

File: jupyterhub/notebook-images/overlays/additional/minimal-notebook-imagestream.yaml [hunk 2]
```diff
@@ -16,11 +16,11 @@ spec:
...
       opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.8.6"}]'
       opendatahub.io/notebook-python-dependencies: '[{"name":"JupyterLab","version": "3.0.16"}, {"name": "Notebook","version": "6.4.0"}]'
-      openshift.io/imported-from: quay.io/thoth-station/s2i-minimal-py38-notebook
+      openshift.io/imported-from: quay.io/modh/odh-minimal-notebook-container
       opendatahub.io/default-image: "true"
     from:
       kind: DockerImage
-      name: quay.io/thoth-station/s2i-minimal-py38-notebook:v0.0.15
-    name: "v0.0.15"
+      name: quay.io/modh/odh-minimal-notebook-container:latest
```
[the remark this answers is not in the document]
Comment (tmckayus) [reply jupyterhub/notebook-images/overlays/additional/minimal-notebook-imagestream.yaml:23]: latest is just a place holder, it is replaced with the actual rhods version via "sed" in the downstream build.
this copy of manifests is primarily to support downstream, not upstream.  Upstream, caveat emptor.

[the remark this answers is not in the document]
Comment (harshad16) [reply jupyterhub/notebook-images/overlays/additional/minimal-notebook-imagestream.yaml:23]: ah totally forgot about that :)

**Review by crobby — APPROVED** [review]
Tested, this lgtm

**Pull request merged**